### DISCUSSION TITLE
Can't install 3.0.1 un Ubuntu 22.04.2

### DISCUSSION THREAD

**Discussion by lmhsjackson** [rbenv/ruby-build#2200] [Build failures]
Hi,
relevant info:

- I have updated rbenv and ruby build by git pulling.
- I have installed the recommended environment packages `sudo apt-get install autoconf patch build-essential rustc libssl-dev libyaml-dev libreadline6-dev zlib1g-dev libgmp-dev libncurses5-dev libffi-dev libgdbm6 libgdbm-dev libdb-dev uuid-dev`

The last 10 lines of the log are not great, but these might be helpful:

```
make[2]: Entering directory '/tmp/ruby-build.20230531093728.31637.IXK0qS/ruby-3.0.1/ext/-test-/exception'
compiling fs.c
compiling dataerror.c
linking shared-object -test-/dln/empty.so
make[2]: Leaving directory '/tmp/ruby-build.20230531093728.31637.IXK0qS/ruby-3.0.1/ext/-test-/dln/empty'`

`linking shared-object socket.so
make[2]: Leaving directory '/tmp/ruby-build.20230531093728.31637.IXK0qS/ruby-3.0.1/ext/date'
make[2]: Leaving directory '/tmp/ruby-build.20230531093728.31637.IXK0qS/ruby-3.0.1/ext/socket'
make[2]: Leaving directory '/tmp/ruby-build.20230531093728.31637.IXK0qS/ruby-3.0.1/ext/openssl'
linking shared-object ripper.so
make[2]: Leaving directory '/tmp/ruby-build.20230531093728.31637.IXK0qS/ruby-3.0.1/ext/ripper'
make[2]: Entering directory '/tmp/ruby-build.20230531093728.31637.IXK0qS/ruby-3.0.1'
ruby: error while loading shared libraries: libruby.so.3.0: cannot open shared object file: No such file or directory
make[2]: *** [uncommon.mk:1156: revision.tmp] Error 127
```

Here is the full log:
[ruby-build.20230531093728.31637.log](https://github.com/rbenv/ruby-build/files/11614264/ruby-build.20230531093728.31637.log)

Thanks in advance for your time :)

**Comment by mislav** [comment]
If there is a system ruby version on your machine, can you check if it's functional? 

And if there is a `/usr/lib/libruby.so` or `/usr/lib/libruby.so3.0` file on your system, can you check that there are no broken symlinks?

Otherwise, I'm not sure what could have caused this. We've had one earlier report with this failure, but then the OP closed it by explaining it went away after a system refresh.

**Comment by lmhsjackson** [comment]
which ruby throws `/home/myuser/.rbenv/shims/ruby` but there is a `/usr/bin/ruby` that responds to -v with `ruby 3.0.2p107 (2021-07-07 revision 0db68f0233) [x86_64-linux-gnu]`

I have no `/usr/lib/libruby.so` or `/usr/lib/libruby.so3.0` files on my computer.

**Reply by mislav** [reply]
Can you try seeing what gets printed by:
```
/usr/bin/ruby -rrbconfig -e 'puts RbConfig::CONFIG["LIBRUBY_SO"]'
```
I'm not sure if this is the correct avenue of investigation, but I wonder whether you're getting the error because your system Ruby does not have a shared library.

**Reply by lmhsjackson** [reply]
`libruby-3.0.so.3.0.2`

**Reply by mislav** [reply]
Thanks. I'm not sure what is the conclusion. You could try reinstalling `ruby` and `ruby-dev` packages via apt to see if it changes anything.

@hsbt Have you come across this perhaps?

**Reply by lmhsjackson** [reply]
I had already done that (reinstalling ruby and ruby-dev). It didn't fix the issue.

**Comment by nad2000** [comment]
I ran into exactly the same issue (Ubuntu 22.04.5/jammy). Solved it by creating a link to the current lib (`sudo ln /usr/lib/x86_64-linux-gnu/libruby-3.0.so.3.0.2  /usr/lib/x86_64-linux-gnu/libruby.so.3.0`). Thanks @mislav , @lmhsjackson 

Steps
1. Locate the file exactly:
```
$ find /usr/lib  -name libruby\* -ls

   731005      0 lrwxrwxrwx   1 root     root           20 Mar 11 07:14 /usr/lib/x86_64-linux-gnu/libruby-3.0.so.3.0 -> libruby-3.0.so.3.0.2
   713869      0 lrwxrwxrwx   1 root     root           20 Mar 11 07:14 /usr/lib/x86_64-linux-gnu/libruby-3.0.so -> libruby-3.0.so.3.0.2
   730748   3476 -rw-r--r--   2 root     root      3558064 Mar 11 07:14 /usr/lib/x86_64-linux-gnu/libruby-3.0.so.3.0.2
```
1. Create a new link:
```
sudo ln /usr/lib/x86_64-linux-gnu/libruby-3.0.so.3.0.2  /usr/lib/x86_64-linux-gnu/libruby.so.3.0
```
